nuxeo-features/nuxeo-elasticsearch/nuxeo-elasticsearch-core/src/test/java/org/nuxeo/elasticsearch/test/TestHighlight.java

```diff
@@ -0,0 +1,206 @@
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+import org.elasticsearch.action.search.SearchRequestBuilder;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.nuxeo.ecm.core.api.CoreSession;
+import org.nuxeo.ecm.core.api.DocumentModel;
+import org.nuxeo.ecm.core.api.DocumentModelList;
+import org.nuxeo.ecm.core.api.blobholder.BlobHolder;
+import org.nuxeo.ecm.core.api.impl.blob.StringBlob;
+import org.nuxeo.ecm.core.work.api.WorkManager;
+import org.nuxeo.elasticsearch.api.ElasticSearchAdmin;
+import org.nuxeo.elasticsearch.api.ElasticSearchService;
+import org.nuxeo.elasticsearch.query.NxQueryBuilder;
+import org.nuxeo.runtime.test.runner.Features;
+import org.nuxeo.runtime.test.runner.FeaturesRunner;
+import org.nuxeo.runtime.test.runner.LocalDeploy;
+import org.nuxeo.runtime.transaction.TransactionHelper;
+
+import javax.inject.Inject;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * @since 9.1
+ */
+@RunWith(FeaturesRunner.class)
+@Features({ RepositoryElasticSearchFeature.class })
+@LocalDeploy("org.nuxeo.elasticsearch.core:elasticsearch-test-highlight-contrib.xml")
+public class TestHighlight {
+
+    @Inject
+    protected CoreSession session;
+
+    @Inject
+    protected ElasticSearchService ess;
+
+    @Inject
+    protected ElasticSearchAdmin esa;
+
+    @Inject
+    protected WorkManager workManager;
+
+    public void waitForCompletion() throws Exception {
+        workManager.awaitCompletion(20, TimeUnit.SECONDS);
+        esa.prepareWaitForIndexing().get(20, TimeUnit.SECONDS);
+        esa.refresh();
+    }
+
+    public void startTransaction() {
+        if (!TransactionHelper.isTransactionActive()) {
+            TransactionHelper.startTransaction();
+        }
+    }
+
+    @Before
+    public void setUpMapping() throws Exception {
+        esa.initIndexes(true);
+    }
+
+    @Test
+    public void testHighlight() throws Exception {
+
+        startTransaction();
+        DocumentModel doc = session.createDocumentModel("/", "highlight", "File");
+        doc.setPropertyValue("dc:title", "Search me");
+        BlobHolder holder = doc.getAdapter(BlobHolder.class);
+        holder.setBlob(new StringBlob("you know for search"));
+        session.createDocument(doc);
+        session.save();
+
+        DocumentModel doc2 = session.createDocumentModel("/", "highlight2", "File");
+        BlobHolder holder2 = doc2.getAdapter(BlobHolder.class);
+        holder2.setBlob(new StringBlob("test my search with highlight"));
+        session.createDocument(doc2);
+        session.save();
+
+        TransactionHelper.commitOrRollbackTransaction();
+        waitForCompletion();
+        startTransaction();
+
+        SearchRequestBuilder request = esa.getClient().prepareSearch("nxutest").setTypes("doc");
+        List<String> highlightFields = new ArrayList<>();
```

Comment:
You can use something like this: `List<String> highlightFields = Arrays.asList("dc:title", "ecm:binarytext");`